src/ledger/ledger-rules/witness-validation.md

```diff
@@ -0,0 +1,81 @@
+# Witness Validation
+
+TODO: provide a description of the ledger rules relevant to this domain specific concept.
```

Comment:
I'm afraid you need to resolve this as I don't get the "foot gun" due to the lack of context (and I consider myself a good example of a potential reader for this being familiar with the EUTxO ledger model, but not have the details ready).

Reply:
Perhaps "foot gun" is the wrong term for this particular example. It's something that's easy to get wrong, but technically the logic matches the specification. One would probably assume that `bootstrap_witnesses` contains all the bootstrap witnesses (as I did), but that would be a wrong assumption.

With very specific payloads, you can provide a vkey witness for a bootstrap address.